app/controllers/users/dossiers_controller.rb

```diff
@@ -671,5 +709,26 @@ def siret_params
     def commentaire_params
       params.require(:commentaire).permit(:body, piece_jointe: [])
     end
+
+    def redirect_if_hidden_or_deleted_dossier
+      dossier_id = params[:id]
+      hidden_dossier = current_user.dossiers
+        .unscope(where: :hidden_by_user_at)
+        .where(id: dossier_id)
+        .where.not(hidden_by_user_at: nil)
+        .first
+
+      if hidden_dossier
+        redirect_to corbeille_dossier_path(dossier_id)
+        return
+      end
+
+      deleted_dossier = DeletedDossier.find_by(dossier_id:, user_id: current_user.id)
+
+      if deleted_dossier
```

Comment:
pas besoin d'affectation ici

```suggestion
      if DeletedDossier.find_by(dossier_id:, user_id: current_user.id)
```